modules/standard/Regex.chpl

```diff
@@ -611,14 +619,79 @@ record regex {
 
   proc init=(x: regex(?)) {
     this.exprType = x.exprType;
-    this.home = x.home;
-    this._regex = x._regex;
-    this.complete();
-    on home {
-      qio_regex_retain(_regex);
+    /* always bring the regex local */
+    this.home = here;
+    /* if it's local, retain and avoid the recompile (thread safe) */
+    if (x.home == here) {
+      this._regex = x._regex;
+      this.complete();
+      qio_regex_retain(x._regex);
+    } else {
+      /* otherwise recompile locally */
+      this.complete();
+      var serialized = x._serialize();
+      this._deserialize(serialized);
     }
   }
 
+  pragma "no doc"
+  proc _serialize() {
+    var pattern: exprType;
+    var options: qio_regex_options_t;
+    var _regexCopy = _regex;
+
+    on this.home {
+      /* NOTE we can't reference `this` inside this on block because
+       * it can cause a recursive chpl__serialize loop */
+      var patternTemp: c_string;
+      qio_regex_get_pattern(_regexCopy, patternTemp);
+      defer chpl_here_free(patternTemp:c_void_ptr);
+      if exprType == string then {
+        try! pattern = createStringWithNewBuffer(patternTemp);
+      }
+      else {
+        pattern = createBytesWithNewBuffer(patternTemp);
+      }
+
+      var localOptions: qio_regex_options_t;
+      qio_regex_get_options(_regexCopy, localOptions);
+      options = localOptions;
+    }
+    return new __serializeHelper(exprType, pattern, options);
+  }
+
+  pragma "no doc"
+  proc _deserialize(in data) {
+    qio_regex_create_compile(data.pattern.c_str(),
+                             data.pattern.numBytes,
+                             data.options,
+                             this._regex);
+
+  }
+
+  pragma "no doc"
+  proc chpl__serialize() {
+    return _serialize();
+  }
+
+  pragma "no doc"
+  proc type chpl__deserialize(data) {
+    var ret:regex(exprType);
+    ret._deserialize(data);
+    return ret;
+  }
+
+  pragma "no doc"
+  proc getOptions() {
```

Comment:
Unused function now?

Reply:
Hmm, I see this is used in the test. It seems odd to have a function just for test purposes. I might instead have the test check for options indirectly? Like have a regex that ignores case and make sure we're doing case insensitive matching correctly, instead of manually checking the options or something?

Reply:
Used in the test currently to verify that the options get RVF'd appropriately